api/handlers.go

```diff
@@ -1127,9 +1190,23 @@ func (si *ServerImplementation) SearchForTransactions(ctx echo.Context, params g
 		return indexerError(ctx, fmt.Errorf("%s: %w", errTransactionSearch, err))
 	}
 
+	// Check if we have more results than requested (indicating more pages exist)
+	var nextToken *string
+	if len(txns) > int(requestedLimit) {
+		// More results exist, so we need to provide a next token
+		// The fetchTransactions function handles next token generation internally
+		nextToken = strPtr(next)
+		// Truncate results to only return the requested amount
+		txns = txns[:requestedLimit]
+	} else if len(txns) > 0 && next != "" {
```

Comment:
I don't under stand why there's `if len(txns) > 0 && next != ""`.  If the first thing didn't happen, `nextToken should be nilled unconditionally, I think.

Reply:
Yup, dropping - its codegen that I missed fixing.